QUARKUS-3004.md

```diff
@@ -0,0 +1,33 @@
+# QUARKUS-3004: ARM support for JVM mode on OCP
+
+Jira: https://issues.redhat.com/browse/QUARKUS-3004
+
+The goal of this feature is to provide production support for Red Hat build of Quarkus running in JVM mode on OpenShift
+on most commonly used server side 64-bit ARM architectures.
+
+## Scope of the testing
+To verify that RHBQ 3 properly functions on ARM achitectures in JVM mode, the OpenShift test suite will have to pass 
```

Comment:
_the OpenShift test suite will have to pass_

You are referring to the https://github.com/quarkus-qe/quarkus-openshift-test-suite, right? It reminds me _fingers crossed_ approach. I tried to understand what is the main reason why we should support (that is **test**) different processor architectures and I can see 2: detect performance issues (GC, CPU-intensive operations,  startup and throughput performance) and functionality (e.g. instructions for AES encryption, multi-thread programming). Can this be tested by this project?

IMO in order to comply ticket description _To have full ARM support for Quarkus by the end of the year, we need the Ghost release to have been tested using a UBI8 OpenJDK 17 Aarch64 container. The purpose is mainly to identify issues, but the issues may be moved to be fixed in a subsequent release._ I think we need to run smoke tests with Quarkus Test Suite (in OpenShift), or improve support scope definition.

Reply:
https://github.com/quarkus-qe/quarkus-test-suite, but +1 to be more specific in the text

What to detect ... focus is on functional issues and issues related to unavailable aarch64 native libraries within jar files and unavailable images for aarch64

Performance - that's a good point too, but we are not hunting the numbers. StartStopTS is not targeting OpenShift. So maybe just some ad-hoc check of boot time in logs ...

Reply:
> What to detect ... focus is on functional issues and issues related to unavailable aarch64 native libraries within jar files and unavailable images for aarch64

That makes sense, but I didn't see it in the TP, neither I saw which one these are. I'd like TP to me more specific, but from short searching, it is not clear to me which parts of TS does that (which tests).

Reply:
Again, I've changed the wording to make this more clear - hopefully. To support RHBQ Ghost on ARM in the same scope on OCP on ARM64 as on x86_64.

About performance testing - this is a good point, however for now, performance testing is not in our hands.

As for functionality - it's not strictly about what either @michalvavrik + @rsvoboda mention. In EAP, we've managed to find some weird behaviors in JDK on HP-UX causing issues. The testing is just to have a level of confidence that what we signed off on works.

Reply:
> To support RHBQ Ghost on ARM in the same scope on OCP on ARM64 as on x86_64 - WDYT?

Plan is clear to me now, thanks.